test/tests/rackhd20/test_rackhd20_api_notifications.py

```diff
@@ -0,0 +1,265 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+Author(s):
+'''
+
+from time import sleep
+import Queue
+import flogging
+import logging
+import pika
+import unittest
+import threading
+import fit_common
+from nose.plugins.attrib import attr
+logs = flogging.get_loggers()
+amqp_queue = Queue.Queue(maxsize=0)
+from pymongo import MongoClient
+from bson.objectid import ObjectId
+
+IP = None
+
+
+class AmqpWorker(threading.Thread):
+
+    def __init__(self, exchange_name, topic_routing_key, external_callback, timeout=10):
+        threading.Thread.__init__(self)
+        pika_logger = logging.getLogger('pika')
+        if fit_common.VERBOSITY >= 8:
+            pika_logger.setLevel(logging.DEBUG)
+        elif fit_common.VERBOSITY >= 4:
+            pika_logger.setLevel(logging.WARNING)
+        else:
+            pika_logger.setLevel(logging.ERROR)
+        if fit_common.API_PORT == "9090":
+            amqp_port = fit_common.fitports()['amqp-vagrant']
+            MONGO_PORT = 37017
+        else:
+            amqp_port = fit_common.fitports()['amqp']
+            MONGO_PORT = 27017
+            logs.debug_1("{0}".format(MONGO_PORT))
+
+        self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=fit_common.fitargs()["rackhd_host"],
+                                                                            port=amqp_port))
+        self.channel = self.connection.channel()
+        result = self.channel.queue_declare(exclusive=True)
+        queue_name = result.method.queue
+        self.channel.queue_bind(exchange=exchange_name, queue=queue_name, routing_key=topic_routing_key)
+        self.channel.basic_consume(external_callback, queue=queue_name)
+        self.connection.add_timeout(timeout, self.dispose)
+
+    def dispose(self):
+        logs.debug_3('Pika connection timeout')
+        if self.connection.is_closed is False:
+            self.channel.stop_consuming()
+            logs.debug_1("Attempting to close the Pika connection")
+            self.thread_stop = True
+
+    def run(self):
+        logs.debug_7('start consuming')
+        self.channel.start_consuming()
+
+
+@attr(all=True, regression=False, smoke=True)
+class test_alert_notification(unittest.TestCase):
+    CLIENT = None
+    MONGO_PORT = None
+    CATALOGS_COUNT = None
+    NODES_COUNT = None
+    OBMS_COUNT = None
+
+    @classmethod
+    def setUpClass(self):
+        if fit_common.API_PORT == "9090":
+            MONGO_PORT = 37017
+            IP = "10.0.2.2"  # vagrant's ip for localhost
```

Comment:
See previous comment regarding amqp-port.  We should have values set up in the config file so the script doesn't need to check if we are running internal vs external.
Where does 10.0.2.2 come from - shouldn't it be fit_args().rackhd_host?
Otherwise, this test only runs against that IP looks like.